lib/game_loop.rb

```diff
@@ -1,38 +1,60 @@
+require_relative './game_state'
 require_relative './guess'
-require_relative './input'
-require_relative './dialog'
 
 class GameLoop
 
-	def initialize(game_state, input = Input.new)
+	def initialize(ui, gs = GameState, g = Guess)
 		@valid = true
-		@game_state = game_state
-		@dialog = Dialog.new(game_state)
-		@input = input
+		@ui = ui
+		@gs = gs		##RENAME state: GAME STATE CLASS, G: GUESS CLASS
+		@g = g
 	end
 
 	def play
+		@game_state = @gs.new
+		@ui.new_game_message
 		while (@game_state.remaining_guesses > 0)
-			display_game_dialog
-			guess = create_new_guess
-			check_validity(guess)
-		    save_guess_if_valid(guess)
-			return if check_win(guess)
+			@ui.display_game_dialog(@valid, @game_state)
+			input = @ui.get_user_input
+			if check_for_command(input)
+				call_command(input)
+			else
+				guess = create_new_guess(format_code(input))
+				check_validity(guess)
+			    save_guess_if_valid(guess)
+				break if check_win(guess)
+			end
 		end
+		@ui.game_over(@game_state)
+		play if @ui.play_again?
+		@ui.exit_game
 	end
 
 		private
 
-	def display_game_dialog
-		system("clear")
-		@dialog.invalid_message unless @valid
-		@dialog.available_colors
-		@dialog.past_guesses
-		@dialog.prompt_guess
+	def check_for_command(input)
+		input.include?('/')
 	end
 
-	def create_new_guess
-		Guess.new(@input.get_user_guess, @game_state)
+	def call_command(input)
+		case input
+		when '/HELP'
+			@ui.display_how_to_play
+		when '/RESTART'
+			play if @ui.restart?
+		when '/QUIT'
+			@ui.exit_game if @ui.quit?
+		else
+			@ui.unknown_command
+		end
+	end
+
+	def format_code(input)
+		input.chars
+	end
+
+	def create_new_guess(input)
+		@g.new(input, @game_state)
```

Comment:
similar to my above comment, it might make more sense for this class to take a new game in the initializer instead of taking a game class and inheriting the burden of creating a new instance of it